components/org.wso2.carbon.identity.api.idpmgt.endpoint/src/main/java/org/wso2/carbon/identity/api/idpmgt/endpoint/EndpointUtils.java

```diff
@@ -0,0 +1,794 @@
+import org.apache.commons.logging.Log;
+import org.wso2.carbon.identity.api.idpmgt.IDPMgtBridgeServiceClientException;
+import org.wso2.carbon.identity.api.idpmgt.IdPConstants;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.Exceptions.BadRequestException;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.Exceptions.InternalServerErrorException;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.ApplicationPermissionDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.ClaimConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.ClaimDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.ClaimMappingDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.ErrorDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.FederatedAuthenticatorConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.IdPDetailDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.IdentityProviderPropertyDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.JustInTimeProvisioningConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.LocalRoleDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.PermissionsAndRoleConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.PropertyDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.ProvisioningConnectorConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.RoleMappingDTO;
+import org.wso2.carbon.identity.application.common.model.ApplicationPermission;
+import org.wso2.carbon.identity.application.common.model.Claim;
+import org.wso2.carbon.identity.application.common.model.ClaimConfig;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.IdentityProviderProperty;
+import org.wso2.carbon.identity.application.common.model.JustInTimeProvisioningConfig;
+import org.wso2.carbon.identity.application.common.model.LocalRole;
+import org.wso2.carbon.identity.application.common.model.PermissionsAndRoleConfig;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.model.ProvisioningConnectorConfig;
+import org.wso2.carbon.identity.application.common.model.RoleMapping;
+import org.wso2.carbon.idp.mgt.IdentityProviderManagementException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class EndpointUtils {
+
+    private EndpointUtils() {
+
+    }
+
+    /**
+     * Translate identityProvider to identityProviderDTO
+     *
+     * @param identityProvider identity provider that needs to be translated to identity provider DTO
+     * @return IdPDetailDTO
+     */
+    public static IdPDetailDTO translateIDPToIDPDetail(IdentityProvider identityProvider) {
```

Comment:
This method seems too long. Can you break it down ? ie for setting each part separately with private methods.